benchmarking/runner/env_capture.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import platform
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Any
+
+from loguru import logger
+
+from runner.utils import get_obj_for_json
+
+
+def dump_env(output_path: Path) -> dict[str, Any]:
+    env_data = get_env()
+
+    try:
+        freeze = subprocess.check_output(["pip", "freeze"], text=True, timeout=120)  # noqa: S603, S607
+        freeze_txt_path = output_path / "pip-freeze.txt"
+        freeze_txt_path.write_text(freeze)
+        env_data["pip_freeze_txt"] = freeze_txt_path
+    except Exception as e:  # noqa: BLE001
+        logger.warning(f"Failed to capture pip freeze: {e}")
+    try:
+        # Try micromamba first, then conda as fallback
+        cmd = None
+        if shutil.which("micromamba"):
+            cmd = ["micromamba", "list", "--explicit"]
+        elif shutil.which("conda"):
+            cmd = ["conda", "list", "--explicit"]
```

Comment:
if we have uv i think we won't need this..

i believe pip freeze might be failing because uv is installed without `--seed`